kategory/src/main/kotlin/kategory/free/Free.kt

```diff
@@ -6,6 +6,9 @@ typealias FreeF<S> = HK<Free.F, S>
 fun <S, A> FreeKind<S, A>.ev(): Free<S, A> =
         this as Free<S, A>
 
+fun <M, S, A> FreeKind<S, A>.foldMapM(f: FunctionK<S, M>, MM: Monad<M>): HK<M, A> =
+        (this as Free<S, A>).foldMap(f, MM)
```

Comment:
Should this be `this.ev().foldMap(f, MM)` ? 
Also foldMapM has other semantics in Scala which is foldmapping along with the monad. I believe this should be just foldMap if it does not colides with foldMap in Free